lib/punycode.js

```diff
@@ -194,6 +194,9 @@ const adapt = function(delta, numPoints, firstTime) {
  * @returns {String} The resulting string of Unicode symbols.
  */
 const decode = function(input) {
+  if (typeof input !== 'string' && !Array.isArray(input)) {
+    throw new TypeError('Argument must be a string or an array');
```

Comment:
For what reason are arrays allowed? As far as I can tell only the empty array would work as `input`.